### PR TITLE
Handle PXE boot devices order when VM is restarted.

### PR DESCRIPTION
**Pull request by rawagner**

### CODE REVIEW

**Comment by rawagner** [comment]
requires https://github.com/kubevirt/web-ui-components/pull/95

File: frontend/public/kubevirt/components/modals/restart-vm-modal.jsx [hunk 1]
```diff
@@ -3,18 +3,31 @@ import PropTypes from 'prop-types';
...
+    this._killVmi = this.killVmi.bind(this);
   }
 
   submit(event) {
     event.preventDefault();
+
+    const pxePatch = getPxeBootPatch(this.props.resource);
+    if (pxePatch.length > 0) {
+      const pxePromise = k8sPatch(this.props.kind, this.props.resource, pxePatch);
+      this.handlePromise(pxePromise).then(this._killVmi);
```
Comment (suomiy) [inline frontend/public/kubevirt/components/modals/restart-vm-modal.jsx]: I think we are missing fail checks here. Are we planning on some notifications? I think at least error log would suffice for now.

File: frontend/public/kubevirt/components/modals/restart-vm-modal.jsx [hunk 2]
```diff
@@ -3,18 +3,31 @@ import PropTypes from 'prop-types';
...
+    } else {
+      this._killVmi();
+    }
+  }
+
+  killVmi() {
```
Comment (suomiy) [inline frontend/public/kubevirt/components/modals/restart-vm-modal.jsx]: Dialog does not show error or do anything if we fail.
Also disabling the buttons could help when in progress.

File: frontend/public/kubevirt/components/modals/start-stop-vm-modal.jsx [hunk 3]
```diff
@@ -16,36 +16,23 @@ class StartStopVmModal extends PromiseComponent {
...
+      patch.push(...pxePatch);
     }
+
+    patch.push({
+      op: 'replace',
+      path: '/spec/running',
+      value: this.props.start
+    });
+
+    const promise = k8sPatch(this.props.kind, vm, patch);
```
Comment (suomiy) [inline frontend/public/kubevirt/components/modals/start-stop-vm-modal.jsx]: again errors

**Comment by rawagner** [comment]
added fail checks

File: frontend/public/kubevirt/components/modals/restart-vm-modal.jsx [hunk 4]
```diff
@@ -3,23 +3,43 @@ import PropTypes from 'prop-types';
...
 
   submit(event) {
     event.preventDefault();
+
+    const pxePatch = getPxeBootPatch(this.props.resource);
+    if (pxePatch.length > 0) {
+      const pxePromise = k8sPatch(this.props.kind, this.props.resource, pxePatch);
+      this.handlePromise(pxePromise).then(this._killVmi).catch(error => this.setErrorMessage(error));
+    } else {
+      this._killVmi();
```
Comment (suomiy) [inline frontend/public/kubevirt/components/modals/restart-vm-modal.jsx]: missing catch

File: frontend/public/kubevirt/components/modals/restart-vm-modal.jsx [hunk 5]
```diff
@@ -3,23 +3,43 @@ import PropTypes from 'prop-types';
...
+    this.setState({ errorMessage: error.message });
   }
 
   submit(event) {
     event.preventDefault();
+
+    const pxePatch = getPxeBootPatch(this.props.resource);
+    if (pxePatch.length > 0) {
+      const pxePromise = k8sPatch(this.props.kind, this.props.resource, pxePatch);
+      this.handlePromise(pxePromise).then(this._killVmi).catch(error => this.setErrorMessage(error));
```
Comment (suomiy) [inline frontend/public/kubevirt/components/modals/restart-vm-modal.jsx]: unnecessary lambda

File: frontend/public/kubevirt/components/modals/restart-vm-modal.jsx [hunk 6]
```diff
@@ -3,23 +3,43 @@ import PropTypes from 'prop-types';
...
+    }
+  }
+
+  killVmi() {
     const getVmiPromise = k8sGet(VirtualMachineInstanceModel, this.props.resource.metadata.name, this.props.resource.metadata.namespace);
     this.handlePromise(getVmiPromise).then(vmi => {
       const killPromise = k8sKill(VirtualMachineInstanceModel, vmi);
-      this.handlePromise(killPromise).then(this.props.close);
-    });
+      this.handlePromise(killPromise).then(this.props.close).catch(error => this.setErrorMessage(error));
```
Comment (suomiy) [inline frontend/public/kubevirt/components/modals/restart-vm-modal.jsx]: lambdas

File: frontend/public/kubevirt/components/modals/delete-device-modal.jsx [hunk 7]
```diff
@@ -79,7 +82,7 @@ class DeleteDeviceModal extends PromiseComponent {
         Are you sure you want to delete <strong>{device.name}</strong>
         <span> from  <strong>{vm.metadata.name} </strong>?</span>
       </ModalBody>
-      <ModalSubmitFooter errorMessage="" inProgress={false} submitText={'Delete'} cancel={this._cancel} />
+      <ModalSubmitFooter errorMessage={this.state.errorMessage} inProgress={false} submitText={'Delete'} cancel={this._cancel} />
```
Comment (suomiy) [inline frontend/public/kubevirt/components/modals/delete-device-modal.jsx]: Is it possible to use already present `this.state.inProgress`  here?

File: frontend/public/kubevirt/components/modals/delete-device-modal.jsx [hunk 8]
```diff
@@ -14,6 +14,9 @@ class DeleteDeviceModal extends PromiseComponent {
     super(props);
     this._cancel = this.props.cancel.bind(this);
     this._submit = this.submit.bind(this);
+    this.state = {
```
Comment (suomiy) [inline frontend/public/kubevirt/components/modals/delete-device-modal.jsx]: can you please reuse the PromiseComponent state and do not redeclare it?

File: frontend/public/kubevirt/components/modals/restart-vm-modal.jsx [hunk 9]
```diff
@@ -3,23 +3,43 @@ import PropTypes from 'prop-types';
...
 import { VirtualMachineInstanceModel } from '../../models';
+import { getPxeBootPatch } from 'kubevirt-web-ui-components';
 
 class RestartVmModal extends PromiseComponent {
   constructor(props) {
     super(props);
     this._cancel = this.props.cancel.bind(this);
     this._submit = this.submit.bind(this);
+    this._killVmi = this.killVmi.bind(this);
+    this.state = {
```
Comment (suomiy) [inline frontend/public/kubevirt/components/modals/restart-vm-modal.jsx]: same in other components

**Comment by rawagner** [comment]
I didnt notice that errors are already handled in PromiseComponent so I do not need to catch anything, just pass errorMessage to Footer. Added progress too.

**Pull request merged**